pkg/server/plugin/nodeattestor/devid/challenge.go

```diff
@@ -0,0 +1,99 @@
+// (C) Copyright 2021 Hewlett Packard Enterprise Development LP
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+package devid
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"errors"
+	"fmt"
+
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpm2/credactivation"
+	"github.com/spiffe/spire/pkg/common/plugin/devid"
+)
+
+const nonceSize = 20
```

Comment:
Right, this was something we set at the beginning of the implementation and didn't double-check again. Now that you mention it, I believe it makes more sense to use 32 bytes (instead of 20) that give us a more "standard" nonce of 256 bits with enough randomness. In addition, that brings some consistency with other nonce sizes used in the project.